modules/virt-cloning-a-datavolume.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-cloning-a-datavolume-using-smart-cloning.adoc
+
+// `blockstorage` conditionals are used (declared in the "*-block" assembly) to separate content
+
+[id="virt-cloning-a-datavolume_{context}"]
+= Cloning a DataVolume
+
+.Prerequisites
+
+For smart-cloning to occur:
+
+* Your storage provider must support snapshots.
+* The source and target PVCs must be defined to the same namespace.
+* The source and target PVCs must be defined to the same storage class.
+* The VolumeSnapshotClass object must reference the storage class defined to both the source and target PVCs.
+
+If any of these prerequisites are not met, host-assisted cloning occurs when you create a DataVolume with a
+PVC source.
+
+.Procedure
+
+To initiate cloning of a DataVolume:
+
+. Create a YAML file for a DataVolume object that specifies the name of the
+new DataVolume, the name and namespace of the source PVC,
+ifdef::blockstorage[]
+`volumeMode: Block` so that an available block PV is used,
+endif::[]
+and the size of the new DataVolume.
++
+For example:
++
+[source,yaml]
+----
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: <cloner-datavolume> <1>
+spec:
+  source:
+    pvc:
+      namespace: "<source-namespace>" <2>
+      name: "<my-favorite-vm-disk>" <3>
+  pvc:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: <2Gi> <4>
+ifdef::blockstorage[]
+    volumeMode: Block <5>
+endif::[]
+----
+<1> The name of the new DataVolume.
+<2> The namespace where the source PVC exists.
+<3> The name of the source PVC.
+<4> The size of the new DataVolume. You must allocate enough space, or the
+cloning operation fails. The size must be the same as or larger than the source PVC.
+ifdef::blockstorage[]
```

Comment:
@bgaydosrh - what's the usage of blockstorage attribute here (and in other code samples) for? I don't see it defined in the enclosing assembly.

Reply:
I lifted the module code from something written (by Andrew perhaps?) for another assembly with the blockstorage conditional attribute appropriately defined and unset. I have added the definition and un-setting of the conditional in my new assembly for consistency.

Reply:
@bgaydosrh can you link to the new PR to make these fixes?

Using or not using the conditional changes the code samples so this should go through a QE review.

Reply:
Link to new PR to fix changes suggested after merging: #26670 

@kgoldbla can you re-review for for this change in conditionals.